include/deal.II/multigrid/mg_data_out.h

```diff
@@ -0,0 +1,528 @@
+// ---------------------------------------------------------------------
+//
+// Copyright (C) 2019 by the deal.II authors
+//
+// This file is part of the deal.II library.
+//
+// The deal.II library is free software; you can use it, redistribute
+// it, and/or modify it under the terms of the GNU Lesser General
+// Public License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+// The full text of the license can be found in the file LICENSE.md at
+// the top level directory of deal.II.
+//
+// ---------------------------------------------------------------------
+
+#ifndef dealii_mg_data_out_h
+#define dealii_mg_data_out_h
+
+//#include <deal.II/multigrid/mg_base.h>
+#include <deal.II/base/mg_level_object.h>
+
+#include <deal.II/numerics/data_out.h>
+
+DEAL_II_NAMESPACE_OPEN
+
+/*
+
+- test subdomain id
+- test level
+- no dofhandler
+- with dofhandler
+- more than one component
+- parallel
+- matrix-free
+- trilinos
+
+ */
+
+
+/*!@addtogroup mg */
+/*@{*/
+
+
+template <int dim, typename DoFHandlerType = DoFHandler<dim>>
+class SingleLevelDataOut : public DataOut<dim, DoFHandlerType>
+{
+public:
+  SingleLevelDataOut()
+    : level_subdomain_id(-1)
+    , level(-1)
+  {}
+
+  SingleLevelDataOut(const unsigned int level_subdomain_id, unsigned int level)
+    : level_subdomain_id(level_subdomain_id)
+    , level(level)
+  {}
+
+  void
+  reinit(const unsigned int level_subdomain_id, unsigned int level)
+  {
+    this->level_subdomain_id = level_subdomain_id;
+    this->level              = level;
+  }
+
+
+  virtual typename DataOut<dim>::cell_iterator
+  first_cell()
+  {
+    typename DataOut<dim>::cell_iterator cell = this->dofs->begin(level);
+    while ((cell != this->dofs->end(level)) &&
+           (cell->level_subdomain_id() != level_subdomain_id))
+      ++cell;
```

Comment:
This is more easily implemented using `std::find_if`. Or maybe also using `FilteredIterator<...>(SubdomainEqualTo(...), dofs->begin(level))`.